test/e2e/gardener/shoot/create_update_delete.go

```diff
@@ -20,178 +20,245 @@ import (
-						expectedNodeLabels := utils.MergeStringMaps(commonNodeLabels)
-						expectedNodeLabels["worker.gardener.cloud/pool"] = workerPool.Name
-						expectedNodeLabels["worker.gardener.cloud/cri-name"] = string(workerPool.CRI.Name)
-						expectedNodeLabels["worker.gardener.cloud/system-components"] = strconv.FormatBool(workerPool.SystemComponents.Allow)
-
-						kubernetesVersion := f.Shoot.Spec.Kubernetes.Version
-						if workerPool.Kubernetes != nil && workerPool.Kubernetes.Version != nil {
-							kubernetesVersion = *workerPool.Kubernetes.Version
+			if !v1beta1helper.IsWorkerless(s.Shoot) {
+				It("Verify worker node labels", func(ctx SpecContext) {
+					commonNodeLabels := utils.MergeStringMaps(s.Shoot.Spec.Provider.Workers[0].Labels)
+					commonNodeLabels["networking.gardener.cloud/node-local-dns-enabled"] = "false"
+					commonNodeLabels["node.kubernetes.io/role"] = "node"
+
+					Eventually(func(g Gomega) {
+						for _, workerPool := range s.Shoot.Spec.Provider.Workers {
+							expectedNodeLabels := utils.MergeStringMaps(commonNodeLabels)
+							expectedNodeLabels["worker.gardener.cloud/pool"] = workerPool.Name
+							expectedNodeLabels["worker.gardener.cloud/cri-name"] = string(workerPool.CRI.Name)
+							expectedNodeLabels["worker.gardener.cloud/system-components"] = strconv.FormatBool(workerPool.SystemComponents.Allow)
+
+							kubernetesVersion := s.Shoot.Spec.Kubernetes.Version
+							if workerPool.Kubernetes != nil && workerPool.Kubernetes.Version != nil {
+								kubernetesVersion = *workerPool.Kubernetes.Version
+							}
+							expectedNodeLabels["worker.gardener.cloud/kubernetes-version"] = kubernetesVersion
+
+							nodeList := &corev1.NodeList{}
+							g.Expect(s.ShootClient.List(ctx, nodeList, client.MatchingLabels{
+								"worker.gardener.cloud/pool": workerPool.Name,
+							})).To(Succeed())
+							g.Expect(nodeList.Items).To(HaveLen(1), "worker pool %s should have exactly one Node", workerPool.Name)
+
+							for key, value := range expectedNodeLabels {
+								g.Expect(nodeList.Items[0].Labels).To(HaveKeyWithValue(key, value), "worker pool %s should have expected labels", workerPool.Name)
+							}
 						}
-						expectedNodeLabels["worker.gardener.cloud/kubernetes-version"] = kubernetesVersion
+					}).Should(Succeed())
+				}, SpecTimeout(time.Minute))
 
-						nodeList := &corev1.NodeList{}
-						g.Expect(shootClient.Client().List(ctx, nodeList, client.MatchingLabels{
-							"worker.gardener.cloud/pool": workerPool.Name,
-						})).To(Succeed())
-						g.Expect(nodeList.Items).To(HaveLen(1), "worker pool %s should have exactly one Node", workerPool.Name)
+				It("Verify reported CIDRs", func(ctx SpecContext) {
+					// For workerless shoots, the status.networking section is not reported. Skip its verification accordingly.
+					Eventually(func(g Gomega) {
+						g.Expect(s.GardenClient.Get(ctx, client.ObjectKeyFromObject(s.Shoot), s.Shoot)).To(Succeed())
```

Comment:
You could simplify this using komega:
```suggestion
						g.Expect(s.GardenKomega.Get(s.Shoot)()).To(Succeed())
```